# Bioinformatics Master's Projects

Welcome to my Bioinformatics Master's Projects repository! This repository is a collection of projects completed during my master's program in Bioinformatics. Each project explores different aspects of bioinformatics, from algorithm development to data analysis and visualization.

---

## Projects Overview

### Suffix Trie Construction
- **Description**: This project focuses on creating a suffix trie from the suffix array of a given DNA sequence. It explores how sequence complexity affects the size and structure of the suffix trie.
- **Key Concepts**: Suffix tries, suffix arrays, sequence complexity.
- **Tools Used**: Python, Graphviz.
- **Directory**: [Suffix Trie Construction](https://github.com/GiorgosKots/Bioinformatics-Master-s-Projects/blob/main/Suffix-Trie-Construction)

### Gibbs Sampler for Motif Discovery
- **Description**: This project implements the Gibbs Sampler algorithm for discovering hidden motifs in DNA sequences. The Gibbs Sampler is a statistical method used to iteratively sample from a probability distribution, which is particularly useful for motif discovery in bioinformatics.
- **Key Concepts**: Gibbs Sampler, Motif Discovery, Position Weight Matrix (PWM)
- **Tools Used**: Python, NumPy, Regex.
- **Directory**: [Motif Discovery (Gibbs Sampler)](https://github.com/GiorgosKots/Bioinformatics-Master-s-Projects/tree/main/Motif-Discovery(Gibbs-Sampler))

### Burrows-Wheeler Transform (BWT)
- **Description**: This project implements the Burrows-Wheeler Transform (BWT) and its inverse (iBWT), a technique used in data compression and genome indexing. The BWT rearranges a string into a form that's easier to compress, and iBWT restores the original sequence. It's widely used in tools like BWA and Bowtie.
- **Key Concepts**: Burrows-Wheeler Transform, inverse BWT, genome indexing, compression.
- **Tools Used**: Python.
- **Directory**: [Burrows-Wheeler Transform](https://github.com/GiorgosKots/Bioinformatics-Master-s-Projects/tree/main/Burrows-Wheeler%20Transform)

### FastA-Inspired Sequence Matching
- **Description**: This project implements two versions of a sequence alignment algorithm inspired by the FastA algorithm for local sequence alignment between biological sequences. It explores efficient methods to find the most similar sequence(s) in a FASTA-formatted database given a query sequence.
- **Key Concepts**: Sequence alignment, k-mer matching, heuristic methods, FastA algorithm.
- **Tools Used**: Python, NumPy, Regex.
- **Directory**: [FastA-Inspired Sequence Matching](https://github.com/GiorgosKots/Bioinformatics-Master-s-Projects/tree/main/FastA-Simplified)

--- 

## Repository Structure

Each project is contained within its own directory, including:
- **Code**: Source code for the project.
- **Example**: Example of the results.
- **README.md**: A brief overview of the project.

---

## Setup Instructions

1. Clone the repository to your local machine.
2. Navigate to the project directory of interest.
3. Follow the instructions in the project's README.md file to set up and run the project.

---

## Contact

For any questions or inquiries, please contact [Kotsopoulos Giorgos] at [kotsgeorge999@gmail.com].
